draft-ietf-quic-transport.md

```diff
@@ -542,68 +545,132 @@ the server supports, as shown below.
-in {{frames}}.  Cleartext packets also include an integrity check, described in
-{{QUIC-TLS}}.
+The client populates the connection ID and packet number fields with randomly
+selected values.  The randomized packet number is incremented for each
+subsequent packet (see {{packet-numbers}}).
+
+The payload of a Client Initial packet consists of a STREAM frame (or frames)
+for stream 0 containing a cryptographic handshake message, plus PADDING frames
+that increase the size of the packet to a minimum size (see {{packetization}}).
+
+The client uses the Client Initial Packet type for any packet that contains an
+initial cryptographic handshake message.  This includes all cases where a new
+packet containing the initial cryptographic message needs to be created, either
+in response to a Version Negotiation packet, in response to a stateless retry
+({{packet-server-stateless}}).
+
+
+### Server Stateless Retry Packet {#packet-server-stateless}
+
+A Server Stateless Retry packet has a type octet of 0x83.  It carries
+cryptographic handshake messages and acknowledgments.  It is used by a server
+that wishes to perform a stateless retry (see {{stateless-retry}}).
+
+The connection ID field in a Server Stateless Retry packet contains a server
+selected value, see {{connection-id}}.
+
+The packet number contains the packet number from the packet that caused it to
+be sent.  This allows a client to verify that the server received its packet.
+
+After receiving a Server Stateless Retry packet, the next cryptographic
+handshake message MUST be sent by the client on stream 0 at an offset of 0.  The
+client discards any buffers it uses for generating retransmissions of handshake
+data and effectively starts the connection anew, keeping only the cryptographic
+handshake state.  The cryptographic handshake is continued, but the transport
+connection is reset.  Continuing the cryptographic handshake is necessary to
+ensure that an attacker cannot force a downgrade of any cryptographic
+parameters.
+
+The client constructs a new Client Cleartext packet
+({{packet-client-cleartext}}) for subsequent cryptographic handshake messages.
+
+The payload of this packet contains STREAM frames and could contain PADDING and
+ACK frames.
+
+
+### Client Cleartext Packet {#packet-client-cleartext}
+
+A Client Cleartext packet has a type octet of 0x84 and is sent when the client
+has received a packet from the server, which will be either a Server Cleartext
+packet or a Version Negotiation packet.
+
+The connection ID field in a Client Cleartext packet contains a connection ID
+that was selected by the server (see {{connection-id}}).
 
+The payload of this packet contains STREAM frames and could contain PADDING and
+ACK frames.  If this packet is sent in response to a Version Negotiation
+({{packet-version}}) or Server Stateless Retry packet
+({{packet-server-stateless}}), it MUST be padded to the minimum packet size for
```

Comment:
OK, so technically the text I have would allow it to be reduced in size as long as it was still above the protocol-defined minimum.  I think that's OK.